--- a/linux/drivers/media/dvb-core/dvbdev.c	2013-11-29 20:16:16.393550000 +0100
+++ b/linux/drivers/media/dvb-core/dvbdev.c	2013-11-29 20:20:38.103436156 +0100
@@ -83,8 +83,11 @@
 			file->f_op = old_fops;
 			goto fail;
 		}
-		if(file->f_op->open)
+		if(file->f_op->open) {
+			mutex_unlock(&dvbdev_mutex);
			err = file->f_op->open(inode,file);
+			mutex_lock(&dvbdev_mutex);
+		}
 		if (err) {
 			fops_put(file->f_op);
 			file->f_op = fops_get(old_fops);

